go/arrow/flight/client.go

```diff
@@ -33,6 +38,7 @@ type Client interface {
 	// Authenticate uses the ClientAuthHandler that was used when creating the client
 	// in order to use the Handshake endpoints of the service.
 	Authenticate(context.Context, ...grpc.CallOption) error
+	AuthenticateBasicToken(context.Context, string, string, ...grpc.CallOption) (context.Context, error)
```

Comment:
I know its obvious, but having var names might make it quicker to verify second and third params are for user and serial